e2e/tests-dfx/remote_generate_binding.bash

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env bats
+
+load ../utils/_
+
+setup() {
+    standard_setup
+}
+
+teardown() {
+    standard_teardown
+}
+
+@test "remote generate-binding succeeds for --all" {
+    install_asset remote_generate_binding/basic
+
+    assert_command dfx remote generate-binding --all
+
+    if [[ ! -f remote.mo ]]; then
+        echo "remote.mo not created"
+        exit 1
+    fi
+
+    if [[ ! -f remote.rs ]]; then
+        echo "remote.rs not created"
+        exit 1
+    fi
+
+    if [[ ! -f remote.ts ]]; then
+        echo "remote.ts not created"
+        exit 1
+    fi
+
+    if [[ ! -f remote.js ]]; then
+        echo "remote.js not created"
+        exit 1
+    fi
+}
+
+@test "remote generate-binding --overwrite succeeds for --all" {
+    install_asset remote_generate_binding/basic
+    echo "to_overwrite" > remote.mo
+    echo "to_overwrite" > remote.rs
+
+    assert_command dfx remote generate-binding --overwrite --all
+
+    # should generate if not present
+    if [[ ! -f remote.ts ]]; then
+        echo "remote.ts not created"
+        exit 1
+    fi
+
+    if [[ ! -f remote.js ]]; then
+        echo "remote.js not created"
+        exit 1
+    fi
+
+    # should overwrite if already present
+    assert_neq "to_overwrite" "$(cat remote.mo)"
+    assert_neq "to_overwrite" "$(cat remote.rs)"
+}
+
+@test "remote generate-binding does not overwrite if file already present" {
+    install_asset remote_generate_binding/basic
+    echo "to_overwrite" > remote.mo
+    echo "to_overwrite" > remote.rs
+
+    assert_command dfx remote generate-binding --all
+
+    # should generate if not present
+    if [[ ! -f remote.ts ]]; then
+        echo "remote.ts not created"
+        exit 1
+    fi
+
+    if [[ ! -f remote.js ]]; then
+        echo "remote.js not created"
+        exit 1
+    fi
+
+    # should not overwrite if already present
+    assert_eq "to_overwrite" "$(cat remote.mo)"
+    assert_eq "to_overwrite" "$(cat remote.rs)"
+}
+
+@test "remote generate-binding succeeds for specific rust canister" {
+    install_asset remote_generate_binding/basic
+
+    assert_command dfx remote generate-binding remote-rust
+
+    if [[ ! -f remote.rs ]]; then 
+        echo "remote.rs not created"
+        exit 1
+    fi
+
+    if [[ -f remote.mo ]]; then 
+        echo "remote.mo created when it shouldn't"
+        exit 1
```

Comment:
bats has specific functions for reporting failures, for example


```suggestion
        batslib_decorate "remote.mo created when it shouldn't"
        fail
```

Reply:
Good suggestion to refactor the tests. But you're actually supposed to pipe into these commands, otherwise the tests hang indefinitely. Correct use example:
```
echo "This is the error message" \
| batslib_decorate "This is the context similar to what we use in rust with .context()" \
| fail
```

I added proper assertions using these two functions. Now the tests are actually readable